packages/fast-components-react-base/src/auto-suggest/auto-suggest.spec.tsx

```diff
@@ -0,0 +1,286 @@
+        expect(() => {
+            shallow(<AutoSuggest listboxId="listboxId" />);
+        }).not.toThrow();
+    });
+
+    test("should implement unhandledProps", (): void => {
+        const unhandledProps: AutoSuggestUnhandledProps = {
+            "aria-label": "label",
+        };
+
+        const rendered: any = shallow(
+            <AutoSuggest listboxId="listboxId" {...unhandledProps} />
+        );
+
+        expect(rendered.first().prop("aria-label")).toEqual("label");
+    });
+
+    test("initial default state is correct", (): void => {
+        const rendered: any = mount(
+            <AutoSuggest listboxId="listboxId">
+                {itemA}
+                {itemB}
+                {itemC}
+            </AutoSuggest>
+        );
+
+        expect(rendered.state("value")).toBe("");
+        expect(rendered.state("focusedItem")).toBe(null);
+        expect(rendered.state("isMenuOpen")).toBe(false);
+    });
+
+    test("default input region attributes are set correctly", (): void => {
+        const rendered: any = mount(
+            <AutoSuggest listboxId="listboxId" label="test-label" disabled={true}>
+                {itemA}
+                {itemB}
+                {itemC}
+            </AutoSuggest>
+        );
+
+        const input: any = rendered.find("input");
+        expect(input.prop("aria-label")).toEqual("test-label");
+        expect(input.prop("aria-autocomplete")).toEqual("both");
+        expect(input.prop("aria-activedescendant")).toEqual(null);
+        expect(input.prop("aria-owns")).toEqual(null);
+        expect(input.prop("aria-controls")).toEqual(null);
+        expect(input.prop("disabled")).toBe(true);
+        expect(input.prop("role")).toEqual("combobox");
+    });
+
+    test("input region attributes change correctly as options list navigated", (): void => {
+        const container: HTMLDivElement = document.createElement("div");
+        document.body.appendChild(container);
+
+        const rendered: any = mount(
+            <AutoSuggest listboxId="listboxId">
+                {itemA}
+                {itemB}
+                {itemC}
+            </AutoSuggest>,
+            { attachTo: container }
+        );
+
+        let input: any = rendered.find("input");
+        input.simulate("keydown", { keyCode: KeyCodes.arrowDown });
+
+        input = rendered.find("input");
+        expect(input.prop("aria-owns")).toEqual("listboxId");
+        expect(input.prop("aria-controls")).toEqual("listboxId");
+        expect(input.prop("aria-activedescendant")).toEqual("a");
+
+        rendered
+            .find('[displayString="a"]')
+            .simulate("keydown", { keyCode: KeyCodes.arrowDown });
+        input = rendered.find("input");
+        expect(input.prop("aria-owns")).toEqual("listboxId");
+        expect(input.prop("aria-controls")).toEqual("listboxId");
+        expect(input.prop("aria-activedescendant")).toEqual("b");
+
+        document.body.removeChild(container);
```

Comment:
👍 - As an FYI, you can also call an enzyme method here for "detach()"

Reply:
ok, thx